test/support/Db.scala

```diff
@@ -33,29 +39,32 @@ object Db extends DatabaseConnection {
   private implicit def slickSetParameterUUID: SetParameter[UUID] =
     SetParameter { case (v, pp) => pp.setObject(v, Types.OTHER) }
 
-  private val truncateVendorsTableAction = sqlu"TRUNCATE TABLE credentials CASCADE "
-
-  private def vendorExistsAction(name: String) =
-    sql"SELECT credential_id FROM candidates WHERE name = $name"
+  private def consumerExistsAction(owner: String) =
+    sql"SELECT id FROM credentials WHERE owner = $owner"
       .as[String]
       .headOption
 
-  private def vendorKeyAction(name: String) =
-    sql"SELECT key FROM credentials cred JOIN candidates can on cred.id = can.credential_id WHERE can.name = $name"
+  private def consumerKeyAction(owner: String) =
+    sql"SELECT key FROM credentials WHERE owner = $owner"
       .as[String]
       .headOption
 
-  private def vendorTokenAction(name: String) =
-    sql"SELECT token FROM credentials cred JOIN candidates can on cred.id = can.credential_id WHERE can.name = $name"
+  private def consumerTokenAction(owner: String) =
+    sql"SELECT token FROM credentials WHERE owner = $owner"
       .as[String]
       .headOption
 
-  private def saveVendorAction(name: String, token: String) = {
+  private def saveConsumerAndCandidatesAction(owner: String, token: String, candidates: Seq[String]) = {
     val credentialId = UUID.randomUUID()
-    val key = generateConsumerKey(name)
-    DBIO.seq(
-      sqlu"INSERT INTO credentials(id, key, token, owner) VALUES ($credentialId, $key, $token, $name)",
-      sqlu"INSERT INTO candidates(credential_id, name) VALUES ($credentialId, $name)"
-    )
+    val key          = generateConsumerKey(owner)
+
+    sqlu"INSERT INTO credentials(id, key, token, owner) VALUES ($credentialId, $key, $token, $owner)"
```

Comment:
Hard to tell if transactions would make sense here. Would it be better to create parts of the candidates, or none at all? I guess it's okay the way it is.

Reply:
It's just a test helper class so not a big thing. The first call has to complete synchronously before the subsequent calls are run in parallel.